kornia/geometry/epipolar/numeric.py

```diff
@@ -24,3 +24,613 @@ def cross_product_matrix(x: torch.Tensor) -> torch.Tensor:
     cross_product_matrix_flat = torch.stack([zeros, -x2, x1, x2, zeros, -x0, -x1, x0, zeros], dim=-1)
     shape_ = x.shape[:-1] + (3, 3)
     return cross_product_matrix_flat.view(*shape_)
+
+
+# Reference
+# Nistér, David. An efficient solution to the five-point relative pose problem. 2004.
+# https://github.com/danini/graph-cut-ransac/blob/master/src/pygcransac/include/
+# estimators/solver_essential_matrix_five_point_nister.h#L108
+
+
+def o1(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+    r"""Multiply two polynomials of the first order.
+
+    Args:
+        a: a first order polynomial for variables :math:`(x,y,z,1)`.
+                b: a first order polynomial for variables :math:`(x,y,z,1)`.
+
+    Returns:
+        degree 2 poly with the order :math:`(x^2, x*y, x*z, x, y^2, y*z, y, z^2, z, 1)`.
+    """
+
+    return torch.stack(
+        [
+            a[:, 0] * b[:, 0],
+            a[:, 0] * b[:, 1] + a[:, 1] * b[:, 0],
+            a[:, 0] * b[:, 2] + a[:, 2] * b[:, 0],
+            a[:, 0] * b[:, 3] + a[:, 3] * b[:, 0],
+            a[:, 1] * b[:, 1],
+            a[:, 1] * b[:, 2] + a[:, 2] * b[:, 1],
+            a[:, 1] * b[:, 3] + a[:, 3] * b[:, 1],
+            a[:, 2] * b[:, 2],
+            a[:, 2] * b[:, 3] + a[:, 3] * b[:, 2],
+            a[:, 3] * b[:, 3],
+        ],
+        dim=-1,
+    )
+
+
+# Reference
+# Nistér, David. An efficient solution to the five-point relative pose problem. 2004.
+# https://github.com/danini/graph-cut-ransac/blob/aae1f40c2e10e31fd2191bac601c53a189673f60/src/pygcransac/
+# include/estimators/solver_essential_matrix_five_point_nister.h#L156
+
+
+def o2(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
```

Comment:
can you give more explicit names to this functions ?

Reply:
Yes! Thanks. Do 'multiply_deg_one_poly' and 'multiply_deg_two__one_poly' look good?